src/sentry/event_manager.py

```diff
@@ -289,7 +262,7 @@ def __init__(
         project_config=None,
         sent_at=None,
     ):
-        self._data = _decode_event(data, content_encoding=content_encoding)
+        self._data = CanonicalKeyDict(data)
```

Comment:
Definitely for a follow-up: We can evaluate if we still need `CanonicalKeyDict` here. Rust handles all normalization, so it's a good time to ensure access to normalized values and get rid of it.